pkg/api/bzz.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package api
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"hash"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	bmtlegacy "github.com/ethersphere/bmt/legacy"
+	"github.com/gorilla/mux"
+	"golang.org/x/crypto/sha3"
+
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+type bzzPostResponse struct {
+	Hash string
+}
+
+func hashFunc() hash.Hash {
+	return sha3.NewLegacyKeccak256()
+}
+
+func (s *server) bzzUploadHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	data, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		s.Logger.Debugf("bzz: read error: %v", err)
+		s.Logger.Error("bzz: read error")
+		jsonhttp.InternalServerError(w, "cannot read request")
+		return
+	}
+
+	p := bmtlegacy.NewTreePool(hashFunc, swarm.SectionSize, bmtlegacy.PoolSize)
```

Comment:
Must be `swarm.Branches` not `swarm.sectionSize`

Reply:
no such field exists. please elaborate

Reply:
https://github.com/ethersphere/bee/blob/master/pkg/swarm/swarm.go#L17